### PR TITLE
Add Dance Party AI Modal Eyes Test

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR adds a UI eyes test for the Dance Party AI Modal. @breville already added a UI test for the AI modal with https://github.com/code-dot-org/code-dot-org/pull/55327 but he noted [in this comment](https://github.com/code-dot-org/code-dot-org/pull/55327#discussion_r1418193905) that the UI test did not check the toggle between code and effect. Therefore, this eyes test includes a check for the toggle to 'Code' in both LTR and RTL languages.

This test also has checks for the AI Modal panel with 25 emojis, three of them being selected in both LTR and RTL languages. It adds an additional check in RTL for the panel with the 25 emojis before any emojis are selected.

The following video is the running of the test but I added red rectangular outlines at points where snapshots are taken. 

https://github.com/code-dot-org/code-dot-org/assets/107423305/9bb9940c-cd22-43a4-a506-10cc926bb718

I added a new level to allthethings cloning and then tweaking the [`dance_ai_customize_effect.level`](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/config/levels/custom/dance/dance_ai_customize_effect.level) used at https://studio.code.org/s/dance-ai-2023/lessons/1/levels/5 so that the effect is already set for consistency in test runs. 

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/config/levels/custom/dance/allthethings HOC Dance AI Customize.level
Comment (sanchitmalhotra126) [inline dashboard/config/levels/custom/dance/allthethings HOC Dance AI Customize.level:1]: For my understanding, why do we need to create a separate allthethings level for this test? Is it just that we always use allthethings for eyes tests?

**Review by sanchitmalhotra126 — APPROVED** [review]
Test looks good to me! Thanks for covering RTL as well!

**Comment by breville** [comment]
Is it best to have a separate UI test, or could we just add some eyes snapshots to the existing `dance_ai_modal.feature`?

File: dashboard/config/levels/custom/dance/allthethings HOC Dance AI Customize.level
Comment (fisher-alice) [reply dashboard/config/levels/custom/dance/allthethings HOC Dance AI Customize.level:1]: I assumed that we want to use allthethings for eyes test from this snippet included in the description at the top of https://studio.code.org/s/allthethings: 'If you are a code.org engineer: this script contains all the level types for UI testing. '. 
I do see in older eyes tests [like this one](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/ui/features/eyes.feature) that other levels are used. We could use the existing Dance Party level but I would need to update the id names of a couple of the blocks like [this one](https://github.com/code-dot-org/code-dot-org/blob/25e111c9a790daef2d771a373a3b37b70f47c2c5/dashboard/config/levels/custom/dance/allthethings%20HOC%20Dance%20AI%20Customize.level#L94). I also removed the video from the level, but we can include the disabling in the url. However - I hesitate to edit the .level file at this point though.

Comment (fisher-alice) [reply dashboard/config/levels/custom/dance/allthethings HOC Dance AI Customize.level:1]: Lmk what you think! @sanchitmalhotra126

**Comment by fisher-alice** [comment]
> Is it best to have a separate UI test, or could we just add some eyes snapshots to the existing `dance_ai_modal.feature`?

I would need to update the .level file since the new level currently added by this PR already has the effect set for the AI block which is already added to setup and whose params are pre-selected. This is so that we have consistency on test runs.
I would also need to add additional steps or scenarios to the existing UI test as well.

I saw that code-tools recently added [`modal_function_editor.feature`](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/ui/features/code_tools/google_blockly/modal_function_editor.feature)  and [`modal_function_editor_eyes.feature`](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/ui/features/code_tools/google_blockly/modal_function_editor_eyes.feature) so I took a similar approach. Lmk what you think - thanks!

File: dashboard/config/levels/custom/dance/allthethings HOC Dance AI Customize.level
Comment (sanchitmalhotra126) [reply dashboard/config/levels/custom/dance/allthethings HOC Dance AI Customize.level:1]: Makes sense! Agreed with not editing existing levels. I was mostly curious if this was a strict requirement, or just a precedent we had.

**Review by breville — APPROVED** [review]

**Pull request merged**